server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/FloatValuesComparatorSource.java

```diff
@@ -61,7 +61,7 @@
  */
 public class FloatValuesComparatorSource extends IndexFieldData.XFieldComparatorSource {
 
-    private final IndexNumericFieldData indexFieldData;
+    protected final IndexNumericFieldData indexFieldData;
```

Comment:
Same here, let keep the `private` but add package protected `IndexNumericFieldData getIndexFieldData()` method to limit the API exposure.

Reply:
makes sense